src/Shared/Utilities/VersionComparer.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    public class VersionComparer : IComparer<List<string>>
+    {
+
+        /// <summary>
+        /// Convert a version (string) to a list of version parts.
+        /// Just splits into "numbers" and "non-numbers".
+        /// </summary>
+        /// <param name="version"></param>
+        public static List<string> Parse(string version)
+        {
+            var parts = new List<string>();
+            var inNumber = false;
+            var curStr = "";
+
+            for (int i = 0; i < version.Length; i++)
+            {
+                var ch = version[i];
+                if (Char.IsNumber(ch) ^ inNumber)
```

Comment:
If you have 12.1.1 doesn't this not work?

- `1` - inNumber = false, Char.IsNumber = true
    - Passes first if, sets curstr to empty.
    - Adds ch to curstr
    - Adds curstr (`1`) to parts
    - Sets inNumber to true
- `2`
    - inNumber = true, Char.IsNumber(ch) is true
    - Skips first if statement
    - adds ch to curstr (now `12`)
    - Passes second if and adds curstr (`12`) to list.  List is now [`1`,`12`].
    - Sets inNumber to true

Reply:
Maybe try this
```csharp
 public static List<string> Parse(string version)
{
    var parts = new List<string>();
    var curStr = string.Empty;

    for (int i = 0; i < version.Length; i++)
    {
        var ch = version[i];
        if (Char.IsNumber(ch))
        {
            curStr += ch;
        }
        else if (ch == '.'){
            parts.Add(curStr);
            curStr = string.Empty;
        }
        else{
            break;
        }
    }
    if (!string.IsNullOrEmpty(curStr)){
        parts.Add(curStr);
    }
    return parts;
}
```

Reply:
I don't this one would work -- we want `a.1.23-rc1` to parse out as `[a, ., 1, ., 23, rc, 1]` so that we can later compare each part separately (e.g.  `a.1.23-rc1` to  `a.1.-rc2`).